src/SAWScript/Interpreter.hs

```diff
@@ -392,9 +393,14 @@ interpretStmt printBinds stmt =
 interpretFile :: FilePath -> Bool {- ^ run main? -} -> TopLevel ()
 interpretFile file runMain = do
   opts <- getOptions
-  stmts <- io $ SAWScript.Import.loadFile opts file
-  mapM_ stmtWithPrint stmts
-  when runMain interpretMain
+  oldPath <- io getCurrentDirectory
+  backtrack oldPath $
```

Comment:
Instead of `backtrack`, I think you can just use [`bracket`](https://hackage.haskell.org/package/base-4.17.0.0/docs/Control-Exception.html#v:bracket) from `Control.Exception`:

```hs
  X.bracket getCurrentDirectory setCurrentDirectory $
  do io $ setCurrentDirectory (takeDirectory file)
     stmts <- io $ SAWScript.Import.loadFile opts file
     mapM_ stmtWithPrint stmts
     when runMain interpretMain
```

Reply:
Ah, I didn't know about `bracket`. `Control.Exception.bracket` is unsuitable here, though, since `stmtWithPrint` and `interpretMain` run in `TopLevel`, not `IO`. I would use `Control.Monad.Catch.bracket`, but it requires that `TopLevel` be `MonadMask` (and unfortunately that instance isn't inherent/can't be derived). Do you think it would be  generally valuable to create such an instance?

Reply:
Ah, good point. Indeed, `TopLevel` can't implement `MonadMask` because `TopLevel` uses a continuation-based monad, which isn't amenable to implementing `MonadMask`'s `mask` method.

Perhaps we should just reimplement `bracket` for `TopLevel` locally and use it here. That should still be possible since `bracket` doesn't require the full power of `mask`.

Reply:
By reimplementing `bracket`, do you mean generalizing my `backtrack` operation?

Reply:
Not just generalizing it, but I also propose that we rename it to something like `bracket` or `bracketTopLevel`. The name `bracket` has a very widely understood connotation in the Haskell world, and that should immediately clue in readers as to its purpose. (It might also be wise to leave a comment saying that we have to reimplement `bracket` for unfortunate technical reasons.)

Reply:
Ah, I see - that does sound wise. I'll do that.

Reply:
I put it in `SAWScript.Value`, reasoning that that seems to contain a lot of `TopLevel`'s interface. Do you think that's appropriate?

Reply:
Yes, that seems like a good place to put `bracketTopLevel`.